pkg/service/healthcheck/metrics_test.go

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2021 ScyllaDB
+
+package healthcheck
+
+import (
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/scylladb/scylla-manager/pkg/util/uuid"
+)
+
+// This test deadlocked when number of metrics exceeded buffered channel length used for collecting metrics.
+// Issue #2843
+func TestRemoveClusterMetricsWhenNumberOfMetricsExceedsDefaultChannelLength_2843(t *testing.T) {
+	clusterID := uuid.MustRandom()
+	metric := prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "scylla_manager",
+		Subsystem: "healthcheck",
+		Name:      "status",
+	}, []string{clusterKey, hostKey})
+	for i := 0; i < 10000; i++ {
```

Comment:
would it deadlock with just buffersize+1?

Reply:
not really, because channel needs to be full in order to trigger deadlock. It's also being drained in parallel so there has to be enough items in a queue to fill the channel before it's drained.

Reply:
previously there was no buffer so I think apply took only one item then tried calling `f` which got stuck on the lock and didn't proceed to next item - at that point collect could try writing one more value but should get stuck on that because nobody would read it anymore. Think that should hold even with the buffer (just +N)